dali/aux/optical_flow/turing_of/optical_flow_turing.cc

```diff
@@ -0,0 +1,169 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/aux/optical_flow/turing_of/optical_flow_turing.h"
+#include <dlfcn.h>
+
+
+namespace dali {
+namespace optical_flow {
+
+namespace {
+
+void VerifySupport(NV_OF_STATUS status) {
+  switch(status) {
+    case NV_OF_SUCCESS:
+      return;
+    case NV_OF_ERR_OF_NOT_AVAILABLE:
+    case NV_OF_ERR_UNSUPPORTED_DEVICE:
+      throw unsupported_exception();
+    default:
+      DALI_FAIL("Optical flow failed, code: " + std::to_string(status));
+  }
+}
+
+}  // namespace
+
+OpticalFlowTuring::OpticalFlowTuring(dali::optical_flow::OpticalFlowParams params, size_t width,
+                                     size_t height, size_t channels) :
+        OpticalFlowAdapter<kernels::ComputeGPU>(params), width_(width), height_(height),
+        channels_(channels), device_(), context_(), stream_(0, true) {
```

Comment:
Wouldn't it be beneficial to allow for different stream than 0?

Reply:
Added stream support